src/parser/shaman/elemental/modules/core/FlameShock.js

```diff
@@ -15,12 +15,24 @@ class FlameShock extends Analyzer {
     enemies: Enemies,
   };
 
+  badLavaBursts = 0;
+
   get uptime() {
     return this.enemies.getBuffUptime(SPELLS.FLAME_SHOCK.id) / this.owner.fightDuration;
   }
 
-  suggestions(when) {
+  on_byPlayer_cast(event) {
+    if(event.ability.guid !== SPELLS.LAVA_BURST.id) {
+      return;
+    }
+
+    const target = this.enemies.getEntity(event);
+    if(target && !target.hasBuff(SPELLS.FLAME_SHOCK.id)){
+      this.badLavaBursts++;
+    }
+  }
 
+  suggestions(when) {
```

Comment:
I'd suggest using the new suggestion format instead of the old one. Example here:
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/a3718bd306252403a5990aaa86d2036f848a7929/src/parser/druid/balance/modules/features/MoonfireUptime.js#L30